### PR TITLE
Remove Nette/PhpGenerator dependency

### PR DESCRIPTION
**Pull request by chrisdunelm**
Two reasons for this: (1) Nette contained a bug in handling 'use <trait>' statements; (2) Nette forced code to be converted from AST to text at an earlier stage than desirable. Therefore our custom AST has been expanded to cover the 'structural' PHP previously handled by Nette.

### CODE REVIEW

File: src/Ast/AST.php [hunk 1]
```diff
@@ -73,9 +117,9 @@ public function toCode(): string
 
     /**
      * Create a PHP variable.
-     *
```
Comment (jskeet) [inline src/Ast/AST.php:76]: It might be worth moving all the whitespace changes into a separate commit, if you have the energy.

**Review by jskeet — APPROVED** [review]
I'm no PHP developer, but the structure looks reasonable to me.

File: src/Ast/AST.php (hunk 1 above)
Comment (chrisdunelm) [reply src/Ast/AST.php:76]: I agree this would be better in a separate commit or PR, but think I'll leave it as-is this time.

**Review by amanda-tarafa — APPROVED** [review]
LGTM structure wise. Easy to follow.
